pkg/console/operator/operator.go

```diff
@@ -145,6 +147,16 @@ func (c *consoleOperator) Sync(obj metav1.Object) error {
 	}
 
 	if err := c.handleSync(operatorConfig, consoleConfig); err != nil {
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorsv1.OperatorCondition{
```

Comment:
You need to unset this on non-failure right?  Also in general you need to guard any status updates to prevent getting your sync loop stuck in an infinite loop.